pkg/client/client.go

```diff
@@ -234,6 +243,81 @@ func (o *OptimizelyClient) decideAll(userContext OptimizelyUserContext, options
 	return o.decideForKeys(userContext, allFlagKeys, options)
 }
 
+// fetchQualifiedSegments fetches all qualified segments for the user context.
+// request is performed asynchronously only when callback is provided
+func (o *OptimizelyClient) fetchQualifiedSegments(userContext *OptimizelyUserContext, options []pkgOdpSegment.OptimizelySegmentOption, callback func(segments []string, err error)) {
+	var err error
+	defer func() {
+		if r := recover(); r != nil {
+			switch t := r.(type) {
+			case error:
+				err = t
+			case string:
+				err = errors.New(t)
+			default:
+				err = errors.New("unexpected error")
+			}
+			o.logger.Error("fetchQualifiedSegments call, optimizely SDK is panicking with the error:", err)
+			o.logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	// on failure, qualifiedSegments should be reset if a previous value exists.
+	userContext.SetQualifiedSegments(nil)
+
+	if _, err = o.getProjectConfig(); err != nil {
```

Comment:
In case user provides an sdkKey, go-sdk allows user to create client as well as userContext while datafile is being downloaded in the background. As for discarding events, the check to ignore events if odp is not integrated is there in the odp manager.